artifact/api.py

```diff
@@ -69,95 +82,152 @@ def map_location(request, map_id):
+
+            if stat in 'OK':
+                result = json_data['results'][0]         
+                latitude = result['geometry']['location']['lat']
+                longitude = result['geometry']['location']['lng']
+                
+                data = {'title': request.data.get('title'),
+                        'map': map_id,
+                        'latitude': latitude,
+                        'longitude': longitude,
+                        'description': request.data.get('description'),
+                        'external_url': request.data.get('externalurl'),
+                        'fileupload': request.data.get('fileupload'),
+                        'created_by_id': logged_in_user_id,
+                        'created_by_full_name': logged_in_user_name,
+                        'modified_by_id': logged_in_user_id,
+                        'date_created': timezone.now(),
+                        'date_modified': timezone.now(),
+                        }
+
+                serializer = MarkersSerializer(data=data)
+                if serializer.is_valid():
+                    serializer.save()
+                    return Response({'latitude': latitude, 'longitude': longitude}, status=status.HTTP_201_CREATED)
+                else:
+                    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+            
+            else:
+                errors = ['Address not found!']
+                return JSONResponse(errors, status=status.HTTP_400_BAD_REQUEST)
+        
         else:
-            logger.debug(serializer.errors)
-            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+            errors = ['You did not enter either an address or lat/long. Try again.']
+            return JSONResponse(errors, status=status.HTTP_400_BAD_REQUEST)
+
 
+# Uploading multiple points via CSV file
 @login_required
-@api_view(['GET','POST'])
+@api_view(['POST'])
 def csv_points(request, map_id):
-    requeststring = request.data[u' filename'].splitlines()
-    logged_in_user_id = request.user.username
-    errors = []
-    for row in requeststring[4:-1]:
-        items = row.split(",")
-        data = {'title': items[0],
-                'map': map_id,
-                'latitude': items[2],
-                'longitude': items[3],
-                'description': items[1],
-                'external_url': items[4],
-                'created_by': logged_in_user_id,
-                'modified_by': logged_in_user_id,
-                'date_created': timezone.now(),
-                'date_modified': timezone.now(),
-                }
-        serializer = MarkersSerializer(data=data)
-        if serializer.is_valid():
-            serializer.save()
-            # return Response(serializer.data, status=status.HTTP_201_CREATED)
+    if request.method == 'POST':
+        LTI_LAUNCH = request.session.get("LTI_LAUNCH").values()[0]
+        logged_in_user_id = LTI_LAUNCH['lis_person_sourcedid']
```

Comment:
I think it's better to use get here like this: LTI_LAUNCH.get('lis_person_sourcedid') 
If 'lis_person_sourcedid' is not in LTI_LAUNCH this will throw a keyerror. If you use get it will return None if it's not present. You can also specify a different return value liek this LTI_LAUNCH.get('lis_person_sourcedid', 'some new return value')